src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -487,19 +487,23 @@ internal static IReadOnlyList<_View> ClearChildren(UIElement view)
 				TRACE($"- renderTransform: {tr.ToMatrix(element.RenderTransformOrigin, element.ActualSize.ToSize())}");
 
 #if __SKIA__
-			var transformToElement = element.TransformToVisual(null);
+			var transformToElement = UIElement.GetTransform(element, null);
 
 			// The maximum region where the current element and its children might draw themselves
 			// This is expressed in the window (absolute) coordinate space.
 			var clippingBounds = element.Visual.ViewBox?.GetRect() is { } rect
-				? transformToElement.TransformBounds(new Rect(rect.Left, rect.Top, rect.Width, rect.Height))
+				? transformToElement.Transform(rect)
 				: Rect.Infinite;
+			if (element.Visual.Clip?.GetRect(element.Visual) is { } clip)
+			{
+				clippingBounds = clippingBounds.IntersectWith(transformToElement.Transform(clip)) ?? default;
+			}
```

Comment:
Note: This still does not supports non rectangular clip (like rounded rect). It's sufficient for now, but we will need at some point to reverse the logic to do something like `Visual.HitTest(Point)` (which could be recursive by its own).

Reply:
Better support for hit testing in such scenarios will be brought by https://github.com/unoplatform/uno/pull/16201. It's still not a `Visual.HitTest` though.